cachi2/core/package_managers/gomod.py

```diff
@@ -1141,44 +1139,16 @@ def __exit__(
             super().__exit__(exc, value, tb)
 
 
-def _should_vendor_deps(
-    flags: Iterable[str], app_dir: RootedPath, strict: bool
-) -> Tuple[bool, bool]:
+def _vendoring_exists(app_dir: RootedPath) -> bool:
     """
-    Determine if Cachi2 should vendor dependencies and if it is allowed to make changes.
+    Determine if vendored content exists in the repository.
 
-    This is based on the presence of flags:
-    - gomod-vendor-check => should vendor, can only make changes if vendor dir does not exist
-    - gomod-vendor => should vendor, can make changes
-
-    :param flags: flags from the Cachi2 request
     :param app_dir: absolute path to the app directory
-    :param strict: fail the request if the vendor dir is present but the flags are not used?
     :return: (should vendor: bool, allowed to make changes in the vendor directory: bool)
     :raise PackageRejected: if the vendor dir is present, the flags are not used and we are strict
     """
     vendor = app_dir.join_within_root("vendor").path
-
-    if "gomod-vendor-check" in flags:
-        return True, not vendor.exists()
-    if "gomod-vendor" in flags:
-        return True, True
-
-    if strict and vendor.is_dir():
-        raise PackageRejected(
-            reason=(
-                'The "gomod-vendor" or "gomod-vendor-check" flag must be set when your repository '
-                "has vendored dependencies."
-            ),
-            solution=(
-                "Consider removing the vendor/ directory and letting Cachi2 download dependencies "
-                "instead.\n"
-                "If you do want to keep using vendoring, please pass one of the required flags."
-            ),
-            docs=VENDORING_DOC,
-        )
-
-    return False, False
```

Comment:
Right, but there's the use case when a user does not have a vendor folder checked in, and used `gomod-vendor` to create one during the prefetch. If we pushed the changes as proposed, the vendor folder would no longer be created, and although it does not result in a failure, it can potentially break the build later on.

Reply:
I'd argue this use case is very odd and should be scrapped, my question here is how we want to do it.

Reply:
Oh, missed that one, solely due to:

> I'd argue this use case is very odd and should be scrapped

This was always a user error. The fact we even supported it in the first place and allowed the source tree to be dirtied by us was a mistake, so this is an attempt to correct an inherently flawed behaviour <<<< I'd sell it like this in the release notes. After all it really is odd to let some other tool do the work for you, since, without our tool would not be able to vendor their project, that never was a legitimate use case or mission of this project, we only fetch dependencies, we must never step in the way how user wishes to distribute their project.